src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -121,6 +121,24 @@ object WdlExpression {
           case "UnaryNegation" => for(e <- expression) yield e.unaryMinus.get
           case _ => throw new WdlExpressionException(s"Invalid operator: ${a.getName}")
         }
+      case a: Ast if a.getName == "ArrayLiteral" =>
+        val evaluatedElements = a.getAttribute("values").asInstanceOf[AstList].asScala.toVector map {x =>
+          evaluate(x, lookup, functions, interpolateStrings)
+        }
+        evaluatedElements.partition {_.isSuccess} match {
+          case (_, failures) if failures.nonEmpty =>
+            val message = failures.collect {case f: Failure[_] => f.exception.getMessage}.mkString("\n")
+            throw new WdlExpressionException(s"Could not evaluate expression:\n$message")
+          case (successes, _) =>
+            successes.map{_.get.wdlType}.toSet match {
+              case s:Set[WdlType] if s.isEmpty =>
```

Comment:
Boo, oh well. Not worth worrying about here, I'll play around w/ that idea at some later point to see if something reasonable like that can work.